src/store/governanceSlice.ts

```diff
@@ -137,5 +157,26 @@ export const createGovernanceSlice: StateCreator<
         console.log('error fetching reserves');
       }
     },
+    getTokenNonce: async (user: string, token: string) => {
+      const service = new ERC20_2612Service(getCorrectProvider());
+      const nonce = await service.getNonce({ token, owner: user });
+      return nonce || 0;
+    },
+    delegateTokensBySig: async (args) => {
+      const governanceService = new AaveGovernanceService(getCorrectProvider(), {
+        GOVERNANCE_ADDRESS: governanceConfig.addresses.AAVE_GOVERNANCE_V2,
+        GOVERNANCE_HELPER_ADDRESS: governanceConfig.addresses.AAVE_GOVERNANCE_V2_HELPER,
```

Comment:
The `governanceService` can be initialized once and re-used

Reply:
About this i think we are using this method for every tx, maybe we should address it in other pr (probably moving it from `zustand` to `useMutation` (react-query) and using the service in `SharedDependenciesProvider` since it needs an architecture change and its already on the react query pr.

Reply:
Sounds good :+1: